pgjdbc/src/test/java/org/postgresql/test/jdbc42/UpdateObject310Test.java

```diff
@@ -0,0 +1,127 @@
+  private Connection con;
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][]{
+        {LocalDateTime.now().truncatedTo(ChronoUnit.MICROS), "timestamp_without_time_zone_column", "'2003-01-01'::TIMESTAMP"},
+        {OffsetDateTime.now().withOffsetSameInstant(ZoneOffset.UTC).truncatedTo(ChronoUnit.MICROS), "timestamp_with_time_zone_column", "'2007-06-03'::TIMESTAMP"},
+        {LocalDate.now(), "date_column", "'2003-01-01'::DATE"},
+        {LocalTime.now().truncatedTo(ChronoUnit.MICROS), "time_without_time_zone_column", "'07:23'::TIME WITHOUT TIME ZONE"}
+    });
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    con = TestUtil.openDB();
+    TestUtil.createTableWithPrimaryKey(con, "table1", "timestamp_without_time_zone_column timestamp without time zone,"
+        + "timestamp_with_time_zone_column timestamp with time zone,"
+        + "date_column date,"
+        + "time_without_time_zone_column time without time zone"
+    );
+  }
+
+  @After
+  public void tearDown() throws SQLException {
+    TestUtil.dropTable(con, "table1");
+    TestUtil.closeDB(con);
+  }
+
+  private Object newValue;
+  private String columnName;
+  private String originalValue;
+
+  public UpdateObject310Test(Object newValue, String columnName, String originalValue) {
+    this.newValue = newValue;
+    this.columnName = columnName;
+    this.originalValue = originalValue;
+  }
+
+  /**
+   * Test the behavior of updateObject for timestamp columns.
+   */
+  @Test
+  public void test() throws SQLException {
+    Statement s = con.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE);
+    ResultSet rs = null;
+
+    try {
+      s.executeUpdate("INSERT INTO table1(" + columnName + ") VALUES(" + originalValue + ")");
+      rs = s.executeQuery(TestUtil.selectSQL("table1", "table1_id, " + columnName));
+      rs.next();
+      rs.updateObject(2, newValue);
+      rs.updateRow();
+      rs.close();
+
+      rs = s.executeQuery(TestUtil.selectSQL("table1", columnName));
+      rs.next();
+      assertEquals(newValue, rs.getObject(1, newValue.getClass()));
+    } finally {
+      if (rs != null) {
+        rs.close();
+      }
+      s.close();
+    }
+  }
+
+  /**
+   * Test the behavior of updateObject for timestamp columns.
+   */
+  @Test
+  public void testWrongClass() throws SQLException {
+    Statement s = con.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE);
+    ResultSet rs = null;
+
+    try {
+      s.executeUpdate("INSERT INTO table1(" + columnName + ") VALUES(" + originalValue + ")");
+      rs = s.executeQuery(TestUtil.selectSQL("table1", "table1_id, " + columnName));
+      rs.next();
+      rs.updateObject(2, 23);
+      rs.updateRow();
+      rs.close();
+
+      fail("should have thrown an Exception");
+    } catch (PSQLException e) {
+      assertEquals(PSQLState.DATATYPE_MISMATCH.getState(), e.getSQLState());
```

Comment:
Please add error message and keep stacktrace.

Current implementation would throw "assertion error" and it would hide the original exception, stacktrace, etc.

Reply:
I added error messages, but I don't understand how you want me to keep the stacktrace, I tried looking at other unit tests in the code base without finding an example.